### ISSUE TITLE
Percolator seed is not getting set properly

### ISSUE THREAD

**Issue by wsnoble** [crux-toolkit/crux-toolkit#574]
If you set the `--seed` option for Crux, that value should get passed on to Percolator.  This does not appear to be happening.  For example, if I do

`/crux percolator --seed 7718 --overwrite T sample.target.txt`

then the percolator command line reported in the log file is

`INFO: percolator --results-peptides crux-output/percolator.target.peptides.txt --decoy-results-peptides crux-output/percolator.decoy.peptides.txt --results-psms crux-output/percolator.target.psms.txt --decoy-results-psms crux-output/percolator.decoy.psms.txt --verbose 2 --protein-decoy-pattern decoy_ --seed 1 --subset-max-train 0 --trainFDR 0.01 --testFDR 0.01 --maxiter 10 --search-input auto --no-schema-validation --protein-enzyme trypsin --post-processing-tdc crux-output/make-pin.pin`

It appears that the `--seed` option to call Percolator internally is hard-coded to always be 1.

[percolator.log.txt](https://github.com/crux-toolkit/crux-toolkit/files/8894138/percolator.log.txt)
[sample.target.txt](https://github.com/crux-toolkit/crux-toolkit/files/8894141/sample.target.txt)
[sample.decoy.txt](https://github.com/crux-toolkit/crux-toolkit/files/8894144/sample.decoy.txt)

**Comment by bmx8177** [comment]
The relevant parameter is --percolator-seed.
Andy

    On Monday, June 13, 2022 at 01:10:39 PM PDT, William Stafford Noble ***@***.***> wrote:  
 
 


If you set the --seed option for Crux, that value should get passed on to Percolator. This does not appear to be happening. For example, if I do

/crux percolator --seed 7718 --overwrite T sample.target.txt

then the percolator command line reported in the log file is

INFO: percolator --results-peptides crux-output/percolator.target.peptides.txt --decoy-results-peptides crux-output/percolator.decoy.peptides.txt --results-psms crux-output/percolator.target.psms.txt --decoy-results-psms crux-output/percolator.decoy.psms.txt --verbose 2 --protein-decoy-pattern decoy_ --seed 1 --subset-max-train 0 --trainFDR 0.01 --testFDR 0.01 --maxiter 10 --search-input auto --no-schema-validation --protein-enzyme trypsin --post-processing-tdc crux-output/make-pin.pin

It appears that the --seed option to call Percolator internally is hard-coded to always be 1.

percolator.log.txt
sample.target.txt
sample.decoy.txt

**Comment by wsnoble** [comment]
Thanks, Andy. In retrospect, this is a pretty dumb setup because users (like me) can easily get confused. I guess this is because Percolator sets its own seed. I think we should add an error message if someone sets the "--seed" parameter in Percolator, like this:

ERROR: The --seed option is for the tide-index command. To set the random number generator seed in percolator, use the --percolator-seed option.

Andy, do you see any drawback to doing this?

**Comment by bmx8177** [comment]
I think this is a good ideal. However, I do have one concern with how this error message might interact with crux pipeline. Couldn't one set both seeds in crux pipeline? Making this change would make it fail. On the other hand, the crux pipeline documentation does not list "--seed" parameter so maybe a moot point?

**Comment by wsnoble** [comment]
You are right. We should (1) make sure the error message only appears in association with the percolator command and (2) make sure the --seed option is listed for the pipeline command.  Do you want to try to implement this?  Should be straightforward, right? :)  I can code review it.

**Comment by bmx8177** [comment]
(2) is easy. We need to add the tide-index options to crux-pipeline (noticed that none of the options were there).
On the other hand....I don't have a good sense how to do (1). Do you know if there is a way to determine which command is being used (ie percolator v pipeline)? I'm not aware of a way.

**Comment by wsnoble** [comment]
Couldn't you put this check in around line 109 of [this file](https://github.com/crux-toolkit/crux-toolkit/blob/master/src/app/PercolatorApplication.cpp)?

**Comment by bmx8177** [comment]
Yes, I agree that is where I would put this check. Let me rephrase my question to make it clearer.
I am unaware of a variable (or fxn) that tells me what Crux command is being used (eg tide-index and psm-convert). That info is necessary to be able to distinguish whether "crux percolator" was run or "crux pipeline" was run. Are you aware of such a variable?

**Comment by wsnoble** [comment]
Yes, I think I understood the question.  I don't believe such a variable exists.  However, I was under the impression that PercolatorApplication.ccp is only called when the command is "crux percolator," and not by "crux pipeline."  That's why I suggested putting the check in there.  Am I wrong?

**Comment by bmx8177** [comment]
"crux pipeline" can call Percolator as a post-processing step (instead of assign-confidence). By calling Percolator, it uses PercolatorApplication.cpp. Specifically "crux pipeline" calls the main method that uses the pin file as the input (line 117) instead of a PSM file as input (line 43).

**Comment by wsnoble** [comment]
Ah, I didn't realize that.  Perhaps you can use the CruxApplication 'name' field (getName method)?

https://github.com/crux-toolkit/crux-toolkit/blob/master/src/app/CruxApplicationList.cpp

**Comment by bmx8177** [comment]
Unfortunately the 'name' field didn't work as CruxApplication is an abstract class. 

However, I got around the problem in a different way. Patch is attached. Let me know what you think. Note that I made some lint test changes. I also did not add a note to the release notes as this wasn't a bug fix per say. 

[patch.txt](https://github.com/crux-toolkit/crux-toolkit/files/8988919/patch.txt)

**Comment by bmx8177** [comment]
Looks like I had a stray tab in there. It has been fixed. Hopefully it works now.
[patch.txt](https://github.com/crux-toolkit/crux-toolkit/files/9067713/patch.txt)

**Comment by wsnoble** [comment]
OK, I have reviewed this patch, and it looks good. Please go ahead and check it in.

**Comment by wsnoble** [comment]
I agree.
